[FEATURE] Improve slack error condition

itaise opened the pull request:
Hi,
I tried triggering a slack notification from GE with a bad token, and didn't get any error. 
That is because slack api returned status code 200, with an error description in the response body.

Changes proposed in this pull request:
- add to the check on status code, a check if "ok" field in the response is True. 

When running again with this PR, I got the message, as expected :) 
`Request to Slack webhook returned error 200: {"ok":false,"error":"invalid_auth","warning":"missing_charset","response_metadata":{"warnings":["missing_charset"]}}`

Thank you!

itaise committed:
add condition where status code 200 but request not successful

itaise committed:
Merge branch 'develop' into feature/improve-slack-error

itaise committed:
aaa

itaise committed:
Merge remote-tracking branch 'origin/feature/improve-slack-error' into feature/improve-slack-error

itaise committed:
Merge branch 'great-expectations:develop' into feature/improve-slack-error

talagluck reviewed (commented):
Thanks for this, @itaise! To confirm, will a successful response code for a Slack message always have an "ok" message? If that is the case, then once you update the tests for this, this should be set to approve and merge!

itaise committed:
remove test file

itaise committed:
Merge remote-tracking branch 'origin/feature/improve-slack-error' into feature/improve-slack-error

itaise committed:
Merge branch 'great-expectations:develop' into feature/improve-slack-error

itaise committed:
add different condition for web hook and for token

itaise committed:
Merge remote-tracking branch 'origin/feature/improve-slack-error' into feature/improve-slack-error

itaise committed:
reformat

itaise committed:
reformat

itaise committed:
reformat

itaise commented:
Thank you @talagluck, that was a good point - apparently the OK response is different when using webhook vs. token - so I updated the condition for both cases and tested it. also changed the test - so that the Mock will have a json method and appropriate text in it.  
WDYT? thanks!

talagluck committed:
Merge branch 'develop' into feature/improve-slack-error

talagluck reviewed (approved):
LGTM - thanks for the update and contribution!
